snowplow-demo-new/src/main/java/com/snowplowanalytics/snowplow_demo_new/ui/SchemaDetailScreen.kt

```diff
@@ -0,0 +1,146 @@
+package com.snowplowanalytics.snowplow_demo_new.ui
+
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material.*
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.ArrowBack
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.snowplowanalytics.snowplow.event.ScreenView
+import com.snowplowanalytics.snowplow.payload.SelfDescribingJson
+import com.snowplowanalytics.snowplow_demo_new.R
+import com.snowplowanalytics.snowplow_demo_new.data.Tracking
+import java.util.*
+
+
+@Composable
+fun SchemaDetailScreen(
+    vm: SchemaDetailViewModel,
+    schemaUrl: String,
+    onBackButtonClicked: () -> Unit = {}
+) {
+    LaunchedEffect(Unit, block = {
+        vm.getSchemaDescription(schemaUrl)
+        vm.getSchemaJson(schemaUrl)
+    })
+    val schemaParts = vm.processSchemaUrl(schemaUrl)
+    
+    val entity = SelfDescribingJson(
+        "iglu:com.snowplowanalytics.iglu/anything-a/jsonschema/1-0-0", 
+        hashMapOf("name" to schemaParts.name, "vendor" to schemaParts.vendor)
+    )
+    val event = ScreenView("detail", UUID.randomUUID()).entities(listOf(entity))
+
+    // Tracks a ScreenView with attached context entity.
+    // This entity records information about the specific schema being viewed
+    Tracking.tracker()?.track(event)
+    
+    Scaffold(
+        topBar = {
+            TopAppBar(
+                title = { Text(schemaParts.name) },
+                navigationIcon = {
+                    IconButton(onClick = onBackButtonClicked) {
+                        Icon(
+                            imageVector = Icons.Filled.ArrowBack,
+                            contentDescription = stringResource(R.string.back_button)
+                        )
+                    }
+
+                }
+            )
+        }
+    ) { contentPadding ->
+        Box(modifier = Modifier.padding(contentPadding)) {
+            Column(modifier = Modifier.padding(all = 8.dp)) {
+                Spacer(modifier = Modifier.width(40.dp))
+
+                Column(modifier = Modifier.verticalScroll(rememberScrollState())) {
+                    Text(
+                        text = "URL",
+                        modifier = Modifier.padding(4.dp),
+                        style = TextStyle(fontSize = 12.sp)
```

Comment:
Nit: we could make use of the [styles in Material design](https://developer.android.com/jetpack/compose/designsystems/material#typography) instead of defining everything ourselves.

```suggestion
                        style = MaterialTheme.typography.overline
```